java-tools/OsmAndServer/src/main/java/net/osmand/server/api/services/WikiService.java

```diff
@@ -86,10 +87,95 @@ public FeatureCollection getImagesById(long id, double lat, double lon) {
 		return new FeatureCollection(features.toArray(new Feature[0]));
 	}
 	
+	public String getWikiRawDataFromCache(String url) {
+		String query = "SELECT raw_data FROM wiki.imagesrawdata WHERE url = ? LIMIT 1";
+		return jdbcTemplate.query(query, ps -> ps.setString(1, url), rs -> {
+			if (rs.next()) {
+				return rs.getString("raw_data");
+			}
+			return null;
+		});
+	}
+	
+	public void saveWikiRawDataToCache(String url, String rawData) {
+		String query = "INSERT INTO wiki.imagesrawdata (url, raw_data) VALUES (?, ?)";
```

Comment:
Probably better to rename to something more related to wikimedia naming 
commonswiki_cache (pageid, title, data)